src/Microsoft.AspNet.Mvc.Core/Formatters/FormatFilter.cs

```diff
@@ -122,12 +122,22 @@ public void OnResultExecuting(ResultExecutingContext context)
             }
 
             var objectResult = context.Result as ObjectResult;
-            if (objectResult != null)
+            if (objectResult == null)
             {
-                var contentType = _options.FormatterMappings.GetMediaTypeMappingForFormat(format);
-                objectResult.ContentTypes.Clear();
-                objectResult.ContentTypes.Add(contentType);
+                return;
             }
+
+            // If the developer is setting a content type explicitly, then its takes precedence over the user
+            // supplied content type based on format mapping.
```

Comment:
I would say `If the action set single a content type`.

What about `[Produces(...)]` with a single content type? Do you expect that to behave the same way as if the developer set `response.ContentType`?